src/miral/external_client.cpp

```diff
@@ -125,4 +126,26 @@ auto  miral::ExternalClientLauncher::launch_using_x11(std::vector<std::string> c
     return -1;
 }
 
+void miral::ExternalClientLauncher::snapcraft_launch(const std::string& desktop_file) const
+{
+    if (!self->server)
+        throw std::logic_error("Cannot launch apps when server has not started");
```

Comment:
Any reason why this doesn't use `BOOST_THROW_EXCEPTION()` like we do most other places? Also, I believe we're leaning toward braces around single line ifs (as you have just below)